### ISSUE TITLE
Performance anomaly due to high detail/JKE models

### ISSUE THREAD

**Issue by TP555** [jdmclark/jkgfxmod#9]
Hi

i can't confirm a Peformance improve , im get the same low FPS in Mystery of the Sith , Mission 1 ,in the Big Hangar min. 22FPS.

what i have found too is ,  enable_ssao and enable_parallax breaks the FPS much more

Used Mods , just the Neural Textures , and the Mystery of the Sith 2019GFX Mod.

**Comment by jarronimo** [comment]
I can confirm. I'm still getting in the 20s at 4K in Jedi Knight. I don't have this issue at 1080p or 1440p, My GPU utilization is around 30%, so I know it's not taxing the GPU.

**Comment by TP555** [comment]
Re

and how much FPS , do you have in 1080 or 1440p in JK  ?

im get most under 30FPS , 2560x1080  + Fov 120 to Force the Correct Aspect for 21:9 and MipMap Patch to fix the Blury Textures from far Away  !

in JK , i don't saw less FPS Problems , but in Mystery of the Sith , alot

Check when you have too with Mystery of the Sith   !

@jdmclark 

im think too , the OpenGL Api , is here mighty totaly overloaded , try to fix your Mod , with DX11 , DX11 have much more Performance than OpenGL

Or try for the next Release , force the Colors to 16Bit again (Like DGVoodoo2) , mayby is 32Bit too much for the Engine + all Textures / Lightning Effects etc.

DgVoodoo use too DX11 , JK and Mots run very Smooth , with Constant 30FPS (Capped) , perfect to Record this with OBS Studio with 60FPS.

https://www.youtube.com/watch?v=F642t2d2_mo   JK
https://www.youtube.com/watch?v=zwEib_ctqMo&t=1s   Mots

**Comment by jdmclark** [comment]
Thanks @jarronimo and @TP555 for the follow-up.

Unfortunately I am not able to reproduce this issue locally, so I will need more information in order to identify a root cause. Low GPU utilization (30%) suggests something more interesting is happening here. Could you please tell me your:
- System CPU utilization while JK is running (reported by Task Manager)
- JK process CPU utilization
- CPU make and model

For comparison, other than some stuttering while materials first load, on my desktop I get 60-90 FPS in 4k, with 95% GPU utilization and 17% CPU. My desktop isn't anything special (1080 with a 6600k). I certainly wouldn't expect a lower framerate on a 2080 Ti!

@TP555 

Thanks for the suggestions. One thing to keep in mind is that JkGfxMod will always have to do a lot more work than a faithful emulator. I don't intend to leave clear performance bugs like this one unaddressed, but you shouldn't expect the performance to ever match DgVoodoo. If you need DgVoodoo performance, and DgVoodoo works well for you, then you should use that instead.

**Comment by TP555** [comment]
Re

https://www.youtube.com/watch?v=6SiPaC-qpIM

I did Film that , with my Smartphone Cam , so we see no possible Breakes , from some Capture Tools , like OBS Studio !

This was now , without MipMap and Fov 120 Patch , with both , goes the Perf Much more down !

But look when all Models are Gone , the FPS goes higher

my Settings:
    
"resolution": [ 2560, 1080 ],
    "internal_resolution": null,
    "fullscreen": true,
    "hud_scale": 1.0,
    "max_anisotropy": 2.0,
    "enable_ssao": false,               (Improve the Performance)
    "enable_parallax": false,          (Improve the Performance)
    "command": "jkmy.exe"

Internal Resolution , is that the Downscale Res. to keep my Base Res. , and to use 1280x540 DS , or 5120x2160 US from my Base Res. ?

**Comment by jarronimo** [comment]
@jdmclark Sorry for the delay. My system CPU utilization is around 30% and the JK process CPU utilization goes up to 15%. I'm running it on a Ryzen 1500X.

**Comment by TP555** [comment]
Re

@jarronimo

Try without the jkgfx2019.gob file in the Resources Folder , and try with follow Settings but both single , to see what cause a FPS Boost then  :D

"resolution": 3840x2160 or 2560x1440     mayby it is just an 4k issue in your Case  !
"internal_resolution": null,
"fullscreen": true,
"hud_scale": 1.0,
"max_anisotropy": 2.0,
"enable_ssao": false, (Improve the Performance in my Case)
"enable_parallax": false, (Improve the Performance in my Case)
"command": "jk.exe"  or try to rename the jk.exe in jediknight.exe , and set jediknight.exe as Command !

**Comment by mattolenik** [comment]
I also have this problem, even with vanilla JK. I have a Ryzen 3600X and a Radeon 5700 XT, but I get 15-16 FPS in Canyon Oasis at 4K. CPU utilization for JK.exe will be around 7-8%, overall utilization about the same, and with GPU utilization showing 40-45%. I don't mind just running at 1080p (which is 70+), except that I run into #33.

**Comment by TP555** [comment]
Hi

@mattolenik 

That isn't a mod issue , that is a General problem , with AMD Graphics Card , which perform worse than Nvidia with the OpenGL API.

i have the same problem , with a 5700 non XT , im get the same FPS , like a GTX950.
Both can't Hold the 30FPS , in the big Hangar at First Mission to begin. !

That is why the Mod needs , a better API Support , like DX11 or 12 or Vulkan , but not just for AMD Graphics Card Users.

To the OpenGL Problem , you can reproduce this issue , with every OpenGL App.
My Test result with Doom 2016 and the 5700

High Details , with 2560x1080

OpenGL:  i can't hold the 60FPS in the First Mission , min under 60 , max is around 70-75 FPS
Vulkan : Around 150 , max 200FPS. in the First Mission.

And so bad is OpenGL , on AMD !!

#Staysafe all  !

**Comment by GeneralTantor** [comment]
So to fix open GL issues we probably need to use something like VooDoo? right?

**Comment by TP555** [comment]
Hi

DgVoodoo , can just Wrap Glide API From 3DFX , and DX1-8 to 11

**Comment by SirYodaJedi** [comment]
No, part of the reason jkgfxmod exists is so that we don't have to deal with the inherently hacky nature of dgVoodoo. dgVoodoo isn't used for OpenGL, anyway.

Your best bet for improving performance on AMD (without Clark switching this project over to Vulkan) is probably by using a [Windows version of the open source Mesa drivers](https://www.mesa3d.org/systems.html). OpenGL performance in that seems to be better than AMD's proprietary drivers, and will probably be the best option unless there is a good OpenGL → Vulkan wrapper out there that I don't know of.
